### ISSUE TITLE
Failed 'helm test' should return exit code > 0

### ISSUE THREAD

**Issue by wilreichert** [kubernetes/helm#2166]
When doing `helm test` the return code matches the helm exit status not test result status.

For example
```
% helm test banking-ferrit --cleanup 
RUNNING: banking-ferrit-ansible-test
PASSED: banking-ferrit-ansible-test
RUNNING: banking-ferrit-credentials-test
FAILED: banking-ferrit-credentials-test, run `kubectl logs banking-ferrit-credentials-test --namespace mtest1` for more info
% echo $?
0
```

Other than grep'ing for random strings in the test output there doesn't seem to be a good way determine testing results. Perhaps add an optional flag to test  (i.e. --output json) and get structured output that could easily be parsed by a CI system?

**Comment by technosophos** [comment]
I was under the impression that this was fixed in 2.2.1.

/cc @michelleN

**Comment by xmik** [comment]
I am using:
```
$ helm version
Client: &version.Version{SemVer:"v2.2.2", GitCommit:"1b330722aafcb8123114ae51f69d1e884a326f3e", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.2.2", GitCommit:"1b330722aafcb8123114ae51f69d1e884a326f3e", GitTreeState:"clean"}
```
and also see this.

**Comment by wilreichert** [comment]
should have mentioned using the latest release
```
% helm version
Client: &version.Version{SemVer:"v2.2.3", GitCommit:"1402a4d6ec9fb349e17b912e32fe259ca21181e3", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.2.3", GitCommit:"1402a4d6ec9fb349e17b912e32fe259ca21181e3", GitTreeState:"clean"}
```

**Comment by michelleN** [comment]
Hey @wilreichert and @xmik - ya'll are correct. The exit code should be non-zero on failures. I'm looking into this now. Thanks for your patience.

**Comment by michelleN** [comment]
Closing this since it is fixed in #2425.
